frontend/src/lib/errorResponseMapper.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  ErrorResponse,
+  ErrorResponseMessageMapping,
+  MappedErrorResponse,
+} from "@/types/errorResponse"
+
+/**
+ * Determines whether the parameter is an error response. Will return true
+ * if mapping for the error type exists. Note that this will not check if
+ * the extension properties are available for that type as that would make
+ * the check very complicated.
+ *
+ * @param e Candidate for a possible error response
+ * @param mapping Mapping to check the error response. The default mapping
+ *  should already include everything you need, this is only intended for
+ *  testing.
+ * @returns Whether the candidate is an error response
+ */
+export function isErrorResponse(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- Has to be any because fetch errors are always any
+  e: any,
+  mapping = errorMessages,
+): e is ErrorResponse {
+  return typeof e === "object" && e.type in mapping
+}
+
+/**
+ * For any backend error response, maps the technical details of that response
+ * to a human-readable message in the UI.
+ *
+ * @param e Response received from the backend
+ * @param mapping Mapping to retrieve the messages from. The default mapping should
+ *  already include everything you need, this is only intended for testing.
+ * @returns
+ */
+export function mapErrorResponse(
```

Comment:
I'm a bit unsure how much this whole setup really provides us with type safety and how much is just checking that the type of `ErrorResponseDetail` and the error messages are aligned.

In `mapErrorResponse` TS for example assumes that only the `__fallback__` case ever happens:
![Screenshot 2024-09-06 at 08 50 31](https://github.com/user-attachments/assets/1cd2b141-cc5c-4406-95e6-770ee9e61e57)

I think we could achieve the same using this and keep the type next to its use. This would also allow us to support the children errors in a simpler way and is probably at general a bit simpler:
```ts
export type ErrorResponse2<
  ExtensionFields extends object = object,
  Children = never,
> = {
  type: string
  title?: string
  details?: string
  instance?: string
  errors?: Array<Children>
} & {
  [K in keyof ExtensionFields]?: ExtensionFields[K]
}

const messages = {
  __fallback__: (e) =>
    "Ein unbekannter Fehler ist aufgetreten.",
  "/errors/article-not-found": (e: ErrorResponse2<{ eid: string }>) => ({
    title: `Article with eID ${e.eid} not found`,
  }),
  "/errors/article-of-type-not-found": (e: ErrorResponse2<{ eli: string }>) =>
    `Norm with eli ${e.eli} does not include articles of type ${e.type?.join(",")}`,
  "/errors/norm-not-found": (e) =>
    `Norm with ELI ${e.instance} not found`,
} satisfies {
  [key: string]: (e: ErrorResponse2) => string | MappedErrorResponse
}
```

We would then still need to adjust this function here to do some casts to `any` but that is something we will need anyway (unless we do typechecks at runtime using something like zod).